lib/dat-manager.js

```diff
@@ -0,0 +1,138 @@
+const collect = require('collect-stream')
+const Stats = require('hyperdrive-stats')
+const sub = require('subleveldown')
+const db = require('./db')
+const encoding = require('dat-encoding')
+const Dat = require('dat-js')
+const root = require('./root-dir')
+const fs = require('fs')
+const EventEmitter = require('events')
+
+module.exports = class Manager extends EventEmitter {
+  constructor () {
+    super()
+    this.dats = []
+    this.loadAll((err, _dats) => {
+      if (err) return this.emit('error', err)
+      for (let dat of _dats) this.dats.push(this.watch(dat))
+      this.emit('update')
+    })
+  }
+  watch (dat) {
+    dat.on('update', () => this.emit('update'))
+    return dat
+  }
+  get () {
+    return this.dats
+  }
+  create (dir, cb) {
+    const dat = Dat({ dir })
+    dat.owner = true
+    dat.dir = dir
+    dat.share(err => {
+      if (err) return cb(err)
+      this.persist(dat, err => {
+        if (err) return cb(err)
+        this.dats.push(this.watch(this.activate(dat)))
+        this.emit('update')
```

Comment:
This causes a re render after an archive has been created. Otherwise it would only show up in the list once an internal hyper* event causes an update.